suite-common/connect-init/src/connectInitThunks.ts

```diff
@@ -61,50 +64,26 @@ export const connectInitThunk = createThunk(
 
         const synchronize = getSynchronize();
 
-        const wrappedMethods: Array<keyof typeof TrezorConnect> = [
-            'applySettings',
-            'authenticateDevice',
-            'authorizeCoinjoin',
-            'backupDevice',
-            'cancelCoinjoinAuthorization',
-            'cardanoGetAddress',
-            'cardanoGetPublicKey',
-            'cardanoSignTransaction',
-            'changePin',
-            'checkFirmwareAuthenticity',
-            'cipherKeyValue',
-            'ethereumGetAddress',
-            'ethereumSignTransaction',
-            'getAddress',
-            'getDeviceState',
-            'getFeatures',
-            'getOwnershipProof',
-            'getPublicKey',
-            'pushTransaction',
-            'rebootToBootloader',
-            'recoveryDevice',
-            'resetDevice',
-            'rippleGetAddress',
-            'rippleSignTransaction',
-            'setBusy',
-            'showDeviceTutorial',
-            'signTransaction',
-            'solanaGetAddress',
-            'solanaSignTransaction',
-            'unlockPath',
-            'wipeDevice',
-        ] as const;
+        const wrappedMethods = (
+            Object.keys(TrezorConnect) as Array<keyof typeof TrezorConnect>
+        ).filter(method => {
+            return (
+                ['on', 'off', 'removeAllListeners', 'uiResponse', 'dispose', 'init'].indexOf(
+                    method,
+                ) === -1
+            );
+        });
 
         wrappedMethods.forEach(key => {
             // typescript complains about params and return type, need to be "any"
             const original: any = TrezorConnect[key];
             if (!original) return;
             (TrezorConnect[key] as any) = async (params: any) => {
-                dispatch(lockDevice(true));
-                const result = await synchronize(() => original(params));
-                dispatch(lockDevice(false));
+                if (selectIsDeviceLocked(getState())) {
+                    return await synchronize(() => original(params));
+                }
```

Comment:
Sounds right. Maybe I can sort of go back to the previous behaviour, but still more robust simply by defining a new utility method on connect, something like 
```
TrezorConnect.getMethods
```
Which would return method details, in this case I would be interested in all methods that have `method.useDevice=true` and create a wrapper for them.

Reply:
Yeah, that would be an improvement.

And why do we even need these locks on Suite side? If it is impossible to do two calls in parallel because of Connect, shouldn't Connect handle it? Maybe we can add a new general method param, something like `allowQueue`, which will cause synchronization on Connect's side instead of throwing call in progress?